services/analytics/analyticsServices.ts

```diff
@@ -0,0 +1,18 @@
+'use client'
+
+import { BetterSocialEventTracking } from "analytics/analyticsEventTracking"
+
+const sendAnalytics = (event: BetterSocialEventTracking, data?: object) => {
+  fetch('/api/send-analytics', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      event,
+      data
+    })
+  })
+}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `sendAnalytics` function is missing error handling. If the fetch request fails for any reason (network issues, server errors, etc.), it will fail silently and this could lead to loss of analytics data. Consider adding a `.catch()` block to handle potential errors.

```diff
  const sendAnalytics = (event: BetterSocialEventTracking, data?: object) => {
    fetch('/api/send-analytics', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        event,
        data
      })
    })
+   .catch(error => console.error('Failed to send analytics:', error));
  }
```
Also, you might want to consider returning the Promise from `fetch` so that callers can handle errors or perform actions after the request completes if they need to.

```diff
-  fetch('/api/send-analytics', {
+  return fetch('/api/send-analytics', {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->